CriticalMass/SettingsViewController.swift

```diff
@@ -97,12 +56,34 @@ class SettingsViewController: UITableViewController {
         }
     }
 
+    // MARK: Actions
+
+    @IBAction func gpsCellAction(_ sender: UISwitch) {
+        Preferences.gpsEnabled = sender.isOn
+    }
+
+    @IBAction func darkModeCellAction(_ sender: UISwitch) {
+        let theme: Theme = sender.isOn ? .dark : .light
+        themeController.changeTheme(to: theme)
+        themeController.applyTheme()
+    }
+
     override func numberOfSections(in _: UITableView) -> Int {
         return Section.allCases.count
     }
 
-    override func tableView(_: UITableView, titleForHeaderInSection section: Int) -> String? {
-        return Section.allCases[section].secionTitle
+    override func tableView(_: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let cell = tableView.dequeueReusableHeaderFooterView(withIdentifier: SettingsTableSectionHeader.typeName)
+        let header = cell as! SettingsTableSectionHeader
+        header.titleLabel.text = Section.allCases[section].secionTitle
+        return cell
+    }
+
+    override func tableView(_: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        guard section == Section.info.index else {
```

Comment:
Do we still need this addition after moving gps and night mode to the same section?

Reply:
yes. otherwise it adds an empty one to the other sections or actually it is there anyways but only has a height that makes it visible for the info section